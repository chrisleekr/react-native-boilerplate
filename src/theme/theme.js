import {Dimensions} from 'react-native';
import {colors} from './colors';
const {width, height} = Dimensions.get('window');

// Themes:- Can alter values here. Can only be consumed through Context (see useTheme.js file)
const themes = {
  light: {
    name: 'light',
    barStyle: 'light-content',
    color: '#000000',
    labelColor: '#525E75',
    invertColor: '#ffffff',
    layoutBg: '#f0f0f0',
    headerBg: '#f6f7f8',
    cardBg: '#e4dfa7',
    accent: '#0071ff',
    error: '#B00020',
    errorText: '#ed4337',
    lightBg: '#F5EE9E',
    borderColor: '#B00020',
    buttonActiveColor: '#6200EE',
    buttonDisabledColor: 'rgba(0, 0, 0, 0.26)',
    card: {
      shadowColor: '#000000',
      borderColor: '#B00020'
    },
    ...colors
  },
  dark: {
    name: 'dark',
    barStyle: 'dark-content',
    color: '#ffffff',
    labelColor: '#F1DDBF',
    invertColor: '#000000',
    layoutBg: '#121212',
    headerBg: '#1e1e1e',
    cardBg: '#3f4045',
    accent: '#0071ff',
    error: '#B00020',
    errorText: '#ed4337',
    lightBg: '#F5EE9E',
    borderColor: '#ffffff',
    buttonActiveColor: '#BB86FC',
    buttonDisabledColor: 'rgba(255,255,255, 0.26)',
    card: {
      shadowColor: '#121212',
      borderColor: '#ffffff'
    },
    ...colors
  }
};

const window = {
  // app dimensions
  width,
  height
};

export {themes, window};
