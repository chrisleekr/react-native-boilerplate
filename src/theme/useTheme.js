/**
 * For Dark/Light Mode
 * @exports ThemeContext, ThemeProvider, useTheme
 */

import React, {useState, createContext, useContext} from 'react';
import {Appearance} from 'react-native';

// Import preconfigured themes from theme file
import {themes} from './theme';

// Context
export const ThemeContext = createContext(null);

// Provider to be used in index/App/or top of any parent
export const ThemeProvider = ({children}) => {
  const colorScheme = Appearance.getColorScheme();

  const [theme, setTheme] = useState(themes[colorScheme]);
  return (
    <ThemeContext.Provider value={{theme, setTheme}}>
      {children}
    </ThemeContext.Provider>
  );
};

// useTheme hook for easy access to theme and setTheme
export const useTheme = () => {
  const {theme, setTheme} = useContext(ThemeContext);

  const toggleTheme = v => {
    setTheme(v ? themes.dark : themes.light);
  };

  return {theme, toggleTheme};
};
