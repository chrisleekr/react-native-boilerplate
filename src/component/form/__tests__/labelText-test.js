import 'react-native';
import React from 'react';

import {cleanup, render} from '@testing-library/react-native';

import {ThemeProvider} from '../../../theme/useTheme';

import LabelText from '../labelText';

let rendered;
let element;

describe('LabelText', () => {
  afterEach(cleanup);

  beforeEach(() => {
    rendered = render(
      <ThemeProvider>
        <LabelText
          containerStyle={{backgroundColor: '#000000'}}
          labelStyle={{backgroundColor: '#EFEFEF'}}
          label={'My Label'}
          textStyle={{fontSize: 13}}
          text={'My Text'}
        />
      </ThemeProvider>
    );
  });

  it('matches snapshot', () => {
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  describe('label', () => {
    beforeEach(() => {
      element = rendered.getByText('My Label');
    });

    it('rendered label', () => {
      expect(element).toBeTruthy();
    });
  });

  describe('text', () => {
    beforeEach(() => {
      element = rendered.getByText('My Text');
    });

    it('rendered text', () => {
      expect(element).toBeTruthy();
    });
  });
});
