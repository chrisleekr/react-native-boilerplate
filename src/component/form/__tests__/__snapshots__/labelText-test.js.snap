// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelText matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 10,
      },
      Object {
        "backgroundColor": "#000000",
      },
    ]
  }
  testID="label-text"
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 16,
        },
        Object {
          "backgroundColor": "#EFEFEF",
        },
        Object {
          "color": "#525E75",
        },
      ]
    }
    testID="label"
  >
    My Label
  </Text>
  <Text
    style={
      Array [
        Object {
          "height": 23,
          "width": "100%",
        },
        Object {
          "fontSize": 13,
        },
        Object {
          "color": "#000000",
        },
      ]
    }
    testID="text"
  >
    My Text
  </Text>
</View>
`;
